### PR TITLE
TRUNK-6180: Add locale to context after authentication.

### PR DESCRIPTION
**Pull request by icrc-loliveira**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-6180

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/UserServiceTest.java [hunk 1]
```diff
@@ -1631,6 +1631,15 @@ public void getDefaultLocaleForUser_shouldReturnDefaultLocaleForUserIfConfigured
 		assertEquals(Locale.FRENCH, Context.getUserService().getDefaultLocaleForUser(createdUser));
 	}
 
+	@Test
+	public void getDefaultLocaleForUser_shouldReturnDefaultLocaleForUserIfAlreadySet() {
+		executeDataSet(XML_FILENAME);
+		Context.authenticate("test", "testUser1234");
+
```
Comment (tendomart) [inline api/src/test/java/org/openmrs/api/UserServiceTest.java:1638]: do you have to declare the  Context object several times.

**Comment by tendomart** [comment]
always try running "mvn clean package" before committing.

you have some issues here at

```
Error:    UserServiceTest.saveUserProperties_shouldRemoveAllExistingPropertiesAndAssignNewProperties:1297 expected: <1> but was: <2>
Error:    UserServiceTest.saveUserProperty_shouldAddNewPropertyToExistingUserProperties:1278 expected: <1> but was: <2>
```

**Review by ibacher — CHANGES_REQUESTED** [review]
Thanks @icrc-loliveira! One small adjustment, I think but should be G2G

**Review by ibacher — APPROVED** [review]

**Comment by ibacher** [comment]
@icrc-loliveira Could you try rebasing this on the latest version of master? I think we have a fix for the failing tests.

**Comment by icrc-loliveira** [comment]
Done

File: api/src/test/resources/org/openmrs/api/include/UserServiceTest.xml [hunk 2]
```diff
@@ -63,5 +63,7 @@
   <users user_id="5511" person_id="5511" system_id="511-x" username="test" password="e9fb6ffdc1f903bdd8ae621347d5329762de868347d317dacecebf6e9e73f4d97297cb1237d8edaa02c5a9bd25949b251c2169202236455c23cdf4d828487c4a" salt="b3cc5eea8fdc9f8579b07267f209fe184f2202cc6379922e0b13f2753c641808bbba8b0ecf4d0592db8e5001e57876c506ae4d3579d5274d88ca8f26b7e14d9d" secret_question="a secret" secret_answer="an answer" creator="1" date_created="2008-08-15 15:57:09.0" changed_by="1" date_changed="2008-08-18 11:51:56.0" retired="false" retire_reason="" uuid="658db69a-e53c-11de-8404-001e378eb99e"/>
 
   <user_property user_id="5511" property="some key" property_value="some value" />
-	
+  <user_property user_id="5511" property="defaultLocale" property_value="fr" />
+
+
```
Comment (ibacher) [inline api/src/test/resources/org/openmrs/api/include/UserServiceTest.xml:68]:
```suggestion
```

**Pull request merged**